### ISSUE TITLE
DownloadStateListener#onDownloadFailure is invoked multiple times for a single video failure

### ISSUE THREAD

**Issue by Ilya-Gh** [kaltura/playkit-dtg-android#77]
[DownloadStateListener#onDownloadFailure
](https://github.com/kaltura/playkit-dtg-android/blob/c2c8b56ccf44ca77f327420703fe4f46899eb667/dtglib/src/main/java/com/kaltura/dtg/DownloadStateListener.java#L12) is invoked multiple times for an HLS stream download.

Steps:
1. Start downloading `tears-multi-hls` via sample app
1. Turn off WiFi and mobile data
1. `DownloadStateListener#onDownloadFailure` called multiple times. 

```
2020-05-06 15:25:00.900 29812-30075/com.kaltura.dtg.demo D/MainActivity: onDownloadFailure: <com.kaltura.dtg.DownloadItemImp itemId=tears-multi-hls contentUrl=http://cdntesting.qa.mkaltura.com/p/1091/sp/109100/playManifest/entryId/0_ttfy4uu0/protocol/http/format/applehttp/flavorIds/0_yuv6fomw,0_i414yxdl,0_mwmzcwv0,0_g68ar3sh/a.m3u8?playbackType=offline&clientTag=playkit-dtg%2Fandroid-dev.7572fa6&referrer=TXlBcHBsaWNhdGlvbg%3D%3D&playSessionId=af71bf25-cce8-4443-8da1-6e31f7745701 state=FAILED addedTime=Wed May 06 15:24:24 GMT+03:00 2020 estimatedSizeBytes=161957006 downloadedSizeBytes=7246506>
2020-05-06 15:25:00.901 29812-30075/com.kaltura.dtg.demo D/MainActivity: onDownloadFailure: <com.kaltura.dtg.DownloadItemImp itemId=tears-multi-hls contentUrl=http://cdntesting.qa.mkaltura.com/p/1091/sp/109100/playManifest/entryId/0_ttfy4uu0/protocol/http/format/applehttp/flavorIds/0_yuv6fomw,0_i414yxdl,0_mwmzcwv0,0_g68ar3sh/a.m3u8?playbackType=offline&clientTag=playkit-dtg%2Fandroid-dev.7572fa6&referrer=TXlBcHBsaWNhdGlvbg%3D%3D&playSessionId=af71bf25-cce8-4443-8da1-6e31f7745701 state=FAILED addedTime=Wed May 06 15:24:24 GMT+03:00 2020 estimatedSizeBytes=161957006 downloadedSizeBytes=7246506>
```

That happens because onDownloadFailure is invoked for each HLS segment failure which can happen simultaneously like connection drop: 

https://github.com/kaltura/playkit-dtg-android/blob/f18287d740b3ecb8cf8898c9b60f373ce49b8ac8/dtglib/src/main/java/com/kaltura/dtg/DownloadService.java#L164

I would expect `onDownloadFailure` to be invoked only once for a single HLS playlist download as `startDownload` is called of the whole video and not on individual segments:
https://github.com/kaltura/playkit-dtg-android/blob/be0a6b6066e82dbf4ae43958139cfc7730e29384/dtglib/src/main/java/com/kaltura/dtg/DownloadItemImp.java#L83

**Comment by kaltura-hooks** [comment]
Hi @Ilya-Gh,

Thank for you reporting an issue and helping improve Kaltura!

To get the fastest response time, and help the maintainers review and test your reported issues or suggestions, please ensure that your issue includes the following (please comment with more info if you have not included all this info in your original issue):
 - Is the issue you're experiencing consistent and across platforms? or does it only happens on certain conditions?
    please provide as much details as possible.
 - Which Kaltura deployment you're using: Kaltura SaaS, or self-hosted?
	If self hosted, are you using the RPM or deb install?
 - Packages installed.
	When using RPM, paste the output for:
```
	# rpm -qa \"kaltura*\"
```	
	For deb based systems:
```	
	# dpkg -l \"kaltura-*\"
```
 - If running a self hosted ENV - provide the MySQL server version used
 - If running a self hosted ENV - is this a single all in 1 server or a cluster?
 - If running a self hosted ENV, while making the problematic request, run:
```
	# tail -f /opt/kaltura/log/*.log /opt/kaltura/log/batch/*.log | grep -A 1 -B 1 --color \"ERR:\|PHP\|trace\|CRIT\|\[error\]\"
```
   and paste the output.
 - When relevant, provide any screenshots or screen recordings showing the issue you're experiencing.

For general troubleshooting see:
https://github.com/kaltura/platform-install-packages/blob/Jupiter-10.13.0/doc/kaltura-packages-faq.md#troubleshooting-help

If you only have a general question rather than a bug report, please close this issue and post at:
http://forum.kaltura.org

Thank you in advance,

**Comment by giladna** [comment]
@Ilya-Gh 

we have opened internal ticket and will update on the findings

#77 FEC-10139

**Comment by GouravSna** [comment]
Fix was released [v2.6.3 ](https://github.com/kaltura/playkit-dtg-android/releases/tag/v2.6.3)
